pennylane/tape/__init__.py

```diff
@@ -95,3 +103,8 @@ def disable_tape():
         warnings.warn("Tape mode is not currently enabled.", UserWarning)
     else:
         _mock_stack.pop().close()
+
+
+def tape_mode_active():
+    """Returns whether tape mode is enabled."""
+    return bool(_mock_stack)

```

Comment:
This _should_ be bullet proof, but if there is another bug where the measurements are not correctly mocked when switching to tape mode this logic fails, right?

Reply:
Yeah, you're right. This checks that the mocks were added, but it doesn't check that the mocking was correct.

Perhaps it could be changed to the following?
```suggestion
    return issubclass(qml.QNode, qml.tape.QNode)
```